### PR TITLE
Fix pull return code

### PR DESCRIPTION
**Pull request by jimmyxian**
Fix: https://github.com/docker/swarm/issues/1151
If code logic is OK,  i will fix `import` and `load` in the same way in a separate PR.
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

File: cluster/swarm/cluster.go [hunk 1]
```diff
@@ -307,14 +307,14 @@ func (c *Cluster) Pull(name string, authConfig *dockerclient.AuthConfig, callbac
...
 
 			if callback != nil {
-				callback(engine.Name, "")
+				callback(engine.Name, "", false)
 			}
 			err := engine.Pull(name, authConfig)
 			if callback != nil {
 				if err != nil {
-					callback(engine.Name, err.Error())
+					callback(engine.Name, err.Error(), true)
```
Comment (vieux) [inline cluster/swarm/cluster.go]: maybe the callback could take `status string, err error` so we call

`callback(engine.Name, "", err.Error)` and `callback(engine.Name, "downloaded", nil)`

What do you think ?

Comment (jimmyxian) [inline cluster/swarm/cluster.go]: Yeah, Good idea!

**Comment by jimmyxian** [comment]
ping @vieux 
Check again, thanks.

**Comment by abronan** [comment]
LGTM

**Comment by vieux** [comment]
LGTM

**Pull request merged**